astropy/coordinates/solar_system.py

```diff
@@ -43,79 +43,140 @@
+    .. [2] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/aareadme_de432s.txt
+    .. [3] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/a_old_versions/
     """
-    _value = ('http://naif.jpl.nasa.gov/pub/naif/generic_kernels'
-               '/spk/planets/de430.bsp')
+    _value = 'approximate'
+    _kernel = None
 
     @classmethod
     def validate(cls, value):
-        try:
-            six.moves.urllib.parse.urlparse(value)
-        except:
-            raise ValueError('{} could not be parsed as a URL'.format(value))
-        global KERNEL
-        KERNEL = None
+        # Set up Kernel; if the file is not in cache, this will dowload it.
+        cls.get_kernel(value)
         return value
 
-
-def _get_kernel(*args, **kwargs):
+    @classmethod
+    def get_kernel(cls, value):
+        # ScienceState only ensures the `_value` attribute is up to date,
+        # so we need to be sure any kernel returned is consistent.
+        if cls._kernel is None or cls._kernel.origin != value:
+            kernel = _get_kernel(value)
+            if kernel is not None:
+                kernel.origin = value
+            cls._kernel = kernel
+        return cls._kernel
+
+    @classproperty
+    def kernel(cls):
+        return cls.get_kernel(cls._value)
+
+    @classproperty
+    def bodies(cls):
+        if cls._value is None:
+            return None
+        if cls._value.lower() == 'approximate':
+            return (('earth', 'sun') +
+                    tuple(PLAN94_BODY_NAME_TO_PLANET_INDEX.keys()))
+        else:
+            return tuple(BODY_NAME_TO_KERNEL_SPEC.keys())
+
+
+
+def _get_kernel(value):
     """
     Try importing jplephem, download/retrieve from cache the Satellite Planet
-    Kernel.
+    Kernel corresponding to the given ephemeris.
     """
-    global KERNEL
+    if value is None or value.lower() == 'approximate':
+        return None
+
+    if value.lower() == 'jpl':
+        value = DEFAULT_JPL_EPHEMERIS
+
+    if value.lower() in ('de430', 'de432s'):
+        value = ('http://naif.jpl.nasa.gov/pub/naif/generic_kernels'
+                 '/spk/planets/{:s}.bsp'.format(value.lower()))
+    else:
+        try:
+            six.moves.urllib.parse.urlparse(value)
+        except:
+            raise ValueError('{} was not one of the standard strings and '
+                             'could not be parsed as a URL'.format(value))
 
     try:
         from jplephem.spk import SPK
     except ImportError:
-        raise ImportError("Solar system ephemeris calculations require the "
-                          "jplephem package "
+        raise ImportError("Solar system JPL ephemeris calculations require "
+                          "the jplephem package "
                           "(https://pypi.python.org/pypi/jplephem)")
 
-    if KERNEL is None:
-        KERNEL = SPK.open(download_file(kernel_url.get(), cache=True))
-    return KERNEL
+    return SPK.open(download_file(value, cache=True))
```

Comment:
Doesn't this leave an open file handle ever time you switch kernels? Based on my reading of `jplephem` it looks like `SPK.open` creates a new file handle and then passes it into `SPK` (which passes it into `DAF`). 

So to fix that I think you want to do something like this in the ``if cls._kernel is None or cls._kernel.origin != value`` stanza:
```
cls._kernel.daf.close()
```
?